src/button-group/item-element.tsx

```diff
@@ -0,0 +1,128 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { forwardRef, useEffect, useImperativeHandle, useRef } from 'react';
+
+import { ButtonProps } from '../button/interfaces.js';
+import { ClickDetail, fireCancelableEvent, NonCancelableEventHandler } from '../internal/events';
+import IconButtonItem from './icon-button-item.js';
+import { ButtonGroupProps } from './interfaces.js';
+import MenuDropdownItem from './menu-dropdown-item.js';
+
+import styles from './styles.css.js';
+
+const ItemElement = forwardRef(
+  (
+    {
+      item,
+      dropdownExpandToViewport,
+      tooltip,
+      setTooltip,
+      onItemClick,
+    }: {
+      item: ButtonGroupProps.Item;
+      dropdownExpandToViewport?: boolean;
+      tooltip: null | { item: string; feedback: boolean };
+      setTooltip: (tooltip: null | { item: string; feedback: boolean }) => void;
+      onItemClick?: NonCancelableEventHandler<ButtonGroupProps.ItemClickDetails> | undefined;
+    },
```

Comment:
Suggestion (readability): You could extract the prop definitions in a separtate `ItemElementProps` interface.

Reply:
Extracted ItemElementProps to interfaces.ts